pkg/ldap/datasource.go

```diff
@@ -23,12 +23,243 @@ import (
+// containing serialized request data.
+// The connector responds with the serialize, processed response.
+func (c *ldapClient) ProxyRequest(ctx context.Context, ci *connector.ConnectorInfo, request *Request) (
+	*Response, *framework.Error) {
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, &framework.Error{
+			Message: fmt.Sprintf("Failed to proxy ldap request, %v", err),
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
+		}
+	}
+
+	r := &v1proxy.ProxyRequestMessage{
+		ConnectorId: ci.ID,
+		ClientId:    ci.ClientID,
+		TenantId:    ci.TenantID,
+		Request: &v1proxy.Request{
+			RequestType: &v1proxy.Request_LdapSearchRequest{
+				LdapSearchRequest: &v1proxy.LDAPSearchRequest{
+					Request: string(data),
+				},
+			},
+		},
+	}
+
+	response := &Response{}
+
+	proxyResp, err := c.proxyClient.ProxyRequest(ctx, r)
+	if err != nil {
+		if st, ok := status.FromError(err); ok {
+			code := customerror.GRPCStatusCodeToHTTPStatusCode(st, err)
+			response.StatusCode = code
+
+			return response, nil
+		}
+
+		return nil, customerror.UpdateError(&framework.Error{
+			Message: fmt.Sprintf("Error searching LDAP server: - %v.", err),
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
+		},
+			customerror.WithRequestTimeoutMessage(err, request.RequestTimeoutSeconds),
+		)
+	}
+
+	ldapResp := proxyResp.GetLdapSearchResponse()
+	if ldapResp == nil {
+
+		return nil, &framework.Error{
+			Message: "Error received nil response from the proxy",
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
+		}
+	}
+
+	if err = json.Unmarshal([]byte(ldapResp.Response), response); err != nil {
+
+		return nil, &framework.Error{
+			Message: fmt.Sprintf("Error unmarshalling ldap response from the proxy, %v.", err),
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
+		}
+	}
+
+	return response, nil
+}
+
+// Request sends a paginated search query directly to an LDAP server and returns the processed response.
+func (c *ldapClient) Request(_ context.Context, request *Request) (*Response, *framework.Error) {
+	tlsConfig, sgnlErr := GetTLSConfig(request)
+	if sgnlErr != nil {
+		return nil, sgnlErr
+	}
+
+	conn, err := ldap_v3.DialURL(request.BaseURL, ldap_v3.DialWithTLSConfig(tlsConfig))
+	if err != nil {
+		return nil, &framework.Error{
+			Message: fmt.Sprintf("Failed to dial ldap server - %v", err),
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
+		}
+	}
+
+	// Bind credentials with connection
+	err = conn.Bind(request.BindDN, request.BindPassword)
+	if err != nil {
+		return nil, &framework.Error{
+			Message: fmt.Sprintf("Failed to bind credentials - %v", err),
+			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_DATASOURCE_AUTHENTICATION_FAILED,
+		}
+	}
+	defer conn.Close()
+
+	// Set cursor from the request (if exists)
+	pageControl, ferr := SetPageControl(request)
+	if err != nil {
+		return nil, ferr
+	}
```

Comment:
The wrong var name is being checked, and can we use a more fitting var name? 
```suggestion
	pageControl, ferr := SetPageControl(request)
	if ferr != nil {
		return nil, ferr
	}
```